### ISSUE TITLE
Go vendoring

### ISSUE THREAD

**Issue by ilovezfs** [whyrusleeping/gx#133]
Would you be willing to start using the Go `vendor` directory or one of the other Go vendor systems so that we can avoid Go resource blocks in the Homebrew formulae for `gx` and `gx-go`?

Currently, the `gx` formula has 35 Go resource blocks, and the `gx-go` formula has 15. Checking whether there are updates for any of these resources every time `gx` and `gx-go` are upgraded is not really sustainable in terms of the maintenance time required.

Thanks!

**Comment by whyrusleeping** [comment]
@ilovezfs Yeah, i was thinking I should lock all the deps for gx down. Somewhat ironically this is exactly what I made the tool for, but using the tool to vendor itself creates a bit of a bootstrapping problem i dont want to deal with right now.

**Comment by lgierth** [comment]
> bootstrapping problem

For bootstrapping we could just download it from dist.ipfs.io as we do for e.g. go-ipfs

**Comment by ilovezfs** [comment]
I suggest using `dep ensure` (https://github.com/golang/dep) or `glide install` (https://github.com/Masterminds/glide).

**Comment by ilovezfs** [comment]
Any updates here? We're now formally deprecating `go_resource` blocks in Homebrew formulae.
https://github.com/Homebrew/brew/pull/3401
https://github.com/Homebrew/homebrew-core/issues/21801

**Comment by whyrusleeping** [comment]
@ilovezfs sorry, no update yet. I'm not against using something like `dep ensure` here but havent had the time to act on it.

Does homebrew always do source builds? Would it be okay to just download the binaries from https://dist.ipfs.io/#gx ?

**Comment by yumitsu** [comment]
@whyrusleeping No, homebrew almost always (except for head/devel versio) uses "bottles", which are precompiled versions.
